src/android/com/adobe/phonegap/push/GCMIntentService.java

```diff
@@ -229,6 +229,21 @@ private Bundle normalizeExtras(Context context, Bundle extras) {
                         Log.e(LOG_TAG, "normalizeExtras: JSON exception");
                     }
                 }
+            } else if (key.equals(APPCELERATOR_PAYLOAD)) {
+                try {
+                    String json = extras.getString(key);
+                    JSONObject data = new JSONObject((String) json);
+                    JSONObject android = data.getJSONObject(ANDROID);
+
+                    newExtras.putString(MESSAGE, android.getString(ALERT));
+                    newExtras.putString(TITLE, android.getString(TITLE));
+                    newExtras.putString(ICON, android.getString(ICON));
+                    newExtras.putBoolean(FOREGROUND, false);
+                    newExtras.putBoolean(COLDSTART, PushPlugin.isActive());
+                }
```

Comment:
Yes, I will review the documentation and maybe reuse some existing code (as pointed in the other response)